### ISSUE TITLE
SFT

### ISSUE THREAD

**Issue by qgallouedec** [huggingface/trl#2635]

**Comment by qgallouedec** [comment]
No issue with SFT

```python
from datasets import load_dataset
from transformers import AutoTokenizer
from trl import SFTConfig, SFTTrainer

tokenizer = AutoTokenizer.from_pretrained("trl-internal-testing/tiny-Qwen2ForCausalLM-2.5")

batch_size = 8
gradient_accumulation_steps = 1
output_dir = f"SFT-bsz{batch_size}-grad_acc{gradient_accumulation_steps}"

training_args = SFTConfig(
    output_dir=output_dir,
    per_device_train_batch_size=batch_size,
    gradient_accumulation_steps=gradient_accumulation_steps,
    logging_steps=2,
)

dummy_dataset = load_dataset("trl-internal-testing/zen", "standard_language_modeling")

trainer = SFTTrainer(
    model="trl-internal-testing/tiny-Qwen2ForCausalLM-2.5",
    args=training_args,
    processing_class=tokenizer,
    train_dataset=dummy_dataset["train"],
)

trainer.train()
```

![Image](https://github.com/user-attachments/assets/42c2151e-b8e9-4ba1-b551-482fa8e97cce)

**Comment by swyoon** [comment]
Gradient accumulation of SFTTrainer behaves weirdly when DeepSeek ZeRO 2 is activated. Not sure whether this is a TRL problem though.
https://github.com/huggingface/accelerate/issues/3877
